app/script/main/SingleInstanceHandler.js

```diff
@@ -0,0 +1,148 @@
+      this._startSingleInstanceCheck();
+      return true;
+    }
+
+    /**
+     * Removes the cookie that keeps track of the running instance
+     *
+     * @param {boolean} forceRemoval - do not check that the instance removing it is the current instance
+     * @returns {void} - returns nothing
+     */
+    deregisterInstance(forceRemoval = false) {
+      const singleInstanceCookie = Cookies.getJSON(CONFIG.COOKIE_NAME);
+
+      const isOwnInstanceId = singleInstanceCookie && singleInstanceCookie.appInstanceId === this.instanceId;
+      if (forceRemoval || isOwnInstanceId) {
+        Cookies.remove(CONFIG.COOKIE_NAME);
+      }
+    }
+
+    /**
+     * Adds a listener that will be called whenever another instance boots
+     *
+     * @param {Function} listener - a listener to be executed
+     * @returns {void} - returns nothing
+     */
+    addExtraInstanceStartedListener(listener) {
+      instanceListeners.push(listener);
+      if (instanceListeners.length === 1) {
+        this._startSingleInstanceCheck();
+      }
+    }
+
+    /**
+     * Removes a listener that would have been called whenever another instance boots
+     *
+     * @param {Function} listener - a listener to be removed
+     * @returns {void} - returns nothing
+     */
+    removeExtraInstanceStartedListener(listener) {
+      const index = instanceListeners.indexOf(listener);
+      instanceListeners.splice(index, 1);
+      if (instanceListeners.length === 0) {
+        this._stopSingleInstanceCheck();
+      }
+    }
+
+    /**
+     * Returns true if another instance is running.
+     * Does not check for the id of the running instance and thus cannot be
+     * invoked once the registering of the current instance has been done.
+     *
+     * @param {Function} listener - a listener to be removed
+     * @throws {Error} if the current app has already been registered
+     * @returns {void} - returns nothing
+     */
+    hasOtherRunningInstance() {
+      if (this.instanceRegistered) {
+        throw new Error('Current instance has been registered, cannot check other running instances');
+      }
+
+      return !!Cookies.get(CONFIG.COOKIE_NAME);
+    }
+
+    _isSingleRunningInstance() {
+      if (z.util.Environment.electron) {
+        return true;
+      }
+      const singleInstanceCookie = Cookies.getJSON(CONFIG.COOKIE_NAME);
+
+      return singleInstanceCookie && singleInstanceCookie.appInstanceId === this.instanceId;
+    }
+
+    _checkSingleInstance() {
+      if (!this._isSingleRunningInstance()) {
+        // warn listeners if the app has started in another instance
+        instanceListeners.forEach(listener => listener());
+      }
+    }
+
+    _startSingleInstanceCheck() {
+      if (checkInterval) {
+        this._stopSingleInstanceCheck();
+      }
+      checkInterval = window.setInterval(this._checkSingleInstance.bind(this), CONFIG.INTERVAL);
+    }
+
+    _stopSingleInstanceCheck() {
+      window.clearInterval(checkInterval);
+    }
```

Comment:
While this works I would be a little more explicit:

Lets move the `if (checkInterval)` check to `_stopSingleInstanceCheck` and also reset the `checkInterval` to undefined. I would also rename `checkInterval` to `checkIntervalId` as that is what this is keeping track off.